docs/features/interceptors.md

```diff
@@ -191,7 +191,13 @@ Interceptors are treated like a post-compilation step in this design. Diagnostic
 
 ### User opt-in
 
-Interceptors will require a feature flag during the experimental phase. The flag can be enabled with `/features=InterceptorsPreview` on the command line or `<Features>InterceptorsPreview</Features>` in msbuild.
+To use interceptors, the user project must specify the property `<InterceptorsPreviewNamespaces>`. This is a list of namespaces which are allowed to contain interceptors.
```

Comment:
>This is a list of namespaces which are allowed to contain interceptors. [](http://example.com/codeflow?start=99&length=71)

Just curious, what was the thinking on listing namespaces that can contain interceptors, versus namespaces of APIs that can be intercepted?

Reply:
Paraphrasing from Jared..

1. We didn't like that opting in the interceptable namespace limits what is interceptable to what is known at build time. For example, maybe a generator wants to intercept calls to a method which implements a certain interface method.
2. We didn't want "accidental opt-in" to occur when multiple generators are interested in different original methods in the same namespace.
3. The granularity is meant to be per-generator. A generator should add usually one entry to this list and have that cover every interceptor it is adding to the compilation.